lib/Sema/Constraint.cpp

```diff
@@ -324,8 +327,15 @@ void Constraint::dump(ConstraintSystem *CS) const {
   // Print all type variables as $T0 instead of _ here.
   llvm::SaveAndRestore<bool> X(CS->getASTContext().LangOpts.
                                DebugConstraintSolver, true);
-  
+  // Disabled MSVC warning: only for use within the debugger
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4996)
+#endif
   dump(&CS->getASTContext().SourceMgr);
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif
 }
```

Comment:
So here' the error:  `warning C4996: 'swift::constraints::Constraint::dump': only for use within the debugger`

It comes from a `LLVM_ATTRIBUTE_DEPRECATED` use here: https://github.com/apple/swift/blob/33c322fd226b0345b3771e7a8875f284777d286a/include/swift/AST/AnyFunctionRef.h#L163

> decides not to warn here because it's being called from another deprecated function?

That would surprise me, as, following up the chain of function calls, we would get at least one deprecation warning if this were the behaviour - otherwise all of this would be dead code. I can't find any `#pragma clang diagnostic` disables for `deprecated-declarations`, but maybe its a CMake thing?

That said, I'm really not sure, sorry